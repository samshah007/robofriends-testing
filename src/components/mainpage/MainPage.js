import React from "react";

import CardList from '../cardlist/CardList';
import SearchBox from '../searchbox/SearchBox';
import Scroll from '../scroll/Scroll';
import ErrorBoundry from '../ErrorBoundry';
import Header from '../header/Header';

import './MainPage.css';

class MainPage extends React.Component{
    componentDidMount() {
        this.props.onRequestRobots();
    }
    filteredRobots = () => {
        return this.props.robots.filter(robot => {
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase())
        })
    }
    checkPending = () => {
        return this.props.isPending
    }
    render() {
        const { onSearchChange, isPending } = this.props;
        return (
            <div className='tc'>
            <Header />
            <SearchBox searchChange={onSearchChange}/>
            <Scroll>
                { this.checkPending() ? <h1>Loading</h1> :
                <ErrorBoundry>
                    <CardList robots={this.filteredRobots()} />
                </ErrorBoundry>
                }
            </Scroll>
            </div>
        );
    }
}
export default MainPage;