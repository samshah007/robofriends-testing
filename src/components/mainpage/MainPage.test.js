import { shallow } from "enzyme";
import React from "react";
import MainPage from "./MainPage";

let wrapper;
beforeEach(() => {
    const mockProps = {
        onRequestRobots:jest.fn(),
        isPending:false,
        searchField:'',
        robots:[]
    }
    wrapper = shallow(<MainPage { ...mockProps }></MainPage>)
});
it('expect to render MainPage component', () => {
    expect(wrapper).toMatchSnapshot()
});
it('filters robots correctly', () => {
    const mockProps = {
        onRequestRobots:jest.fn(),
        isPending:false,
        searchField:'a',
        robots:[{
            id:3,
            name:'Shyamal Shah',
            email:'sam@gmail.com'
        }]
    }
    expect(wrapper.instance().filteredRobots()).toEqual([]);
    const wrapper2 = shallow(<MainPage { ...mockProps}></MainPage>);
    expect(wrapper2.instance().filteredRobots()).toEqual([{
        id:3,
        name:'Shyamal Shah',
        email:'sam@gmail.com'
    }]);
});
it('check if isPending working', () => {
    const mockProps3 = {
        onRequestRobots:jest.fn(),
        isPending:true,
        searchField:'a',
        robots:[]
    }
    const wrapper3 = shallow(<MainPage { ...mockProps3}></MainPage>);
    expect(wrapper3.instance().checkPending()).toEqual(true);
});
