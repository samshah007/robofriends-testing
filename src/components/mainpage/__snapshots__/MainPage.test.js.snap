// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render MainPage component 1`] = `ShallowWrapper {}`;
